# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1728159658 -7200
# Parent  d1ead0022ba41035cf116fdd8d6dc95e9307d000
Branding 8: Sign setup.exe and all .exe and .dll files at the correct point in time. Exclude MS DLLs from signing.

diff --git a/browser/installer/windows/sign.py b/browser/installer/windows/sign.py
--- a/browser/installer/windows/sign.py
+++ b/browser/installer/windows/sign.py
@@ -1,26 +1,16 @@
 import subprocess, sys
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Set-AuthenticodeSignature -FilePath instgen\\setup.exe -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert)"],
-    stdout=sys.stdout)
-p.communicate()
-
-p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.exe -File | Where-Object { $_.Name -in @(\"betterbird.exe\", \"pingsender.exe\", \"plugin-container.exe\", \"rnp-cli.exe\", \"rnpkeys.exe\", \"WSEnable.exe\") } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
     stdout=sys.stdout)
 p.communicate()
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird\\uninstall -Filter helper.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\uninstall\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin\\uninstall -Filter helper.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\uninstall\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
     stdout=sys.stdout)
 p.communicate()
 
 p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\bin -Filter betterbird.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    "Get-ChildItem -Path ..\\..\\..\\dist\\bin -Filter *.dll -File | Where-Object { $_.Name -notlike \"vcruntime140*\" -and $_.Name -notlike \"msvcp140.*\" -and $_.Name -notlike \"api-ms-win*\" -and $_.Name -notlike \"d3dcompiler_47.*\" } | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\dist\\bin\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
     stdout=sys.stdout)
 p.communicate()
-
-p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
-    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\betterbird -Filter *.dll -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\betterbird\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
-    stdout=sys.stdout)
-p.communicate()
diff --git a/browser/installer/windows/sign_installer.py b/browser/installer/windows/sign_installer.py
new file mode 100644
--- /dev/null
+++ b/browser/installer/windows/sign_installer.py
@@ -0,0 +1,6 @@
+import subprocess, sys
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Get-ChildItem -Path ..\\..\\..\\..\\dist\\install\\sea -Filter *installer.exe -File | ForEach-Object { Set-AuthenticodeSignature -FilePath ..\\..\\..\\..\\dist\\install\\sea\\$_ -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert) } "],
+    stdout=sys.stdout)
+p.communicate()
diff --git a/browser/installer/windows/sign_setup.py b/browser/installer/windows/sign_setup.py
new file mode 100644
--- /dev/null
+++ b/browser/installer/windows/sign_setup.py
@@ -0,0 +1,6 @@
+import subprocess, sys
+
+p = subprocess.Popen(["C:/Windows/System32/WindowsPowerShell/v1.0/powershell.exe",
+    "Set-AuthenticodeSignature -FilePath instgen\\setup.exe -Certificate (Get-ChildItem -Path Cert:\CurrentUser\My\ -CodeSigningCert)"],
+    stdout=sys.stdout)
+p.communicate()
diff --git a/toolkit/mozapps/installer/packager.mk b/toolkit/mozapps/installer/packager.mk
--- a/toolkit/mozapps/installer/packager.mk
+++ b/toolkit/mozapps/installer/packager.mk
@@ -113,8 +113,12 @@ make-package-internal: prepare-package m
 	@echo 'Compressing...'
 	cd $(DIST) && $(MAKE_PACKAGE)
 
 make-package: FORCE
+ifeq (WINNT,$(OS_ARCH))
+	@echo 'Signing *.exe and *.dll.'
+	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign.py
+endif
 	$(MAKE) make-package-internal
 ifeq (WINNT,$(OS_ARCH))
 ifeq ($(MOZ_PKG_FORMAT),ZIP)
 	$(MAKE) -C windows ZIP_IN='$(ABS_DIST)/$(PACKAGE)' installer
diff --git a/toolkit/mozapps/installer/windows/nsis/makensis.mk b/toolkit/mozapps/installer/windows/nsis/makensis.mk
--- a/toolkit/mozapps/installer/windows/nsis/makensis.mk
+++ b/toolkit/mozapps/installer/windows/nsis/makensis.mk
@@ -62,11 +62,11 @@ ifdef MOZ_STUB_INSTALLER
 endif
 
 ifdef ZIP_IN
 installer:: $(CONFIG_DIR)/setup.exe $(ZIP_IN)
-	@echo 'Signing.'
-	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign.py
-	@echo 'Packaging $(WIN32_INSTALLER_OUT).'
+	@echo 'Signing setup.exe.'
+	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign_setup.py
+	@echo 'Packaging $(ABS_DIST)/$(PKG_INST_PATH)$(PKG_INST_BASENAME).exe'
 	$(NSINSTALL) -D '$(ABS_DIST)/$(PKG_INST_PATH)'
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
 	  -o '$(ABS_DIST)/$(PKG_INST_PATH)$(PKG_INST_BASENAME).exe' \
 	  --package-name '$(MOZ_PKG_DIR)' \
@@ -74,8 +74,10 @@ installer:: $(CONFIG_DIR)/setup.exe $(ZI
 	  --tag $(topsrcdir)/$(MOZ_BUILD_APP)/installer/windows/app.tag \
 	  --setupexe $(CONFIG_DIR)/setup.exe \
 	  --sfx-stub $(SFX_MODULE) \
 	  $(USE_UPX)
+	@echo 'Signing installer.exe.'
+	$(PYTHON3) $(topsrcdir)/browser/installer/windows/sign_installer.py
 ifdef MOZ_STUB_INSTALLER
 	$(PYTHON3) $(MOZILLA_DIR)/mach repackage installer \
 	  -o '$(ABS_DIST)/$(PKG_INST_PATH)$(PKG_STUB_BASENAME).exe' \
 	  --tag $(topsrcdir)/browser/installer/windows/stub.tag \
